test/controllers/posts_controller_test.rb

```diff
@@ -6,16 +6,24 @@ class PostsControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "should get index" do
+    sign_in users(:abdullah)
+
     get posts_url
+
     assert_response :success
```

Comment:
To make this test pass it needed a signed in user. I also put spaces between the three phases of testing

1. setup
2. execution
3. verify

Reply:
Great!